src/pykeen/datasets/base.py

```diff
@@ -228,7 +229,8 @@ def __repr__(self) -> str:  # noqa: D105
 def _name_from_url(url: str) -> str:
     """Get the filename form the end of the URL."""
     parse_result = urlparse(url)
-    return os.path.basename(parse_result.path)
+    path = pathlib.Path(parse_result.path)
```

Comment:
`PurePosixPath`? :smile:  https://docs.python.org/3/library/pathlib.html

Reply:
omg

you are too good for this world

Reply:
Done in 05334d47